src/domain.rs

```diff
@@ -190,7 +191,7 @@ type CrawlRequest<T> = Pin<Box<dyn Future<Output = Result<Response<T>>>>>;
 type RobotsTxtRequest = Pin<Box<dyn Future<Output = Result<RobotsData>>>>;
 
 pub struct AllowedDomain<T> {
-    client: reqwest::Client,
+    client: Box<reqwest_middleware::ClientWithMiddleware>,
```

Comment:
is boxing required here?

Reply:
looking at this, this should be cheap to clone
https://github.com/TrueLayer/reqwest-middleware/blob/main/reqwest-middleware/src/client.rs#L59

Reply:
Yeah, it's not needed. I added it because you were using `&` references as arguments in some functions so the most direct way to "translate" it was to use a box. `ClientWithMiddleware` uses a `Box` and an `Arc` internally, so yeah we should be able to just clone it.

Reply:
Fixed this in the latest commit

Reply:
I'm using the latest commit in my project now. It's working as expected